File: msbuild/tests/NativeExtensionEmbedding/native/NativeTodayExtension/TodayViewController.swift
//
//  TodayViewController.swift
//  NativeTodayExtension
//
//  Created by Chris Hamons on 7/7/20.
//  Copyright © 2020 Chris Hamons. All rights reserved.
//

import UIKit
import NotificationCenter

class TodayViewController: UIViewController, NCWidgetProviding {
    override func viewDidLoad() {
        super.viewDidLoad()
    }
        
    func widgetPerformUpdate(completionHandler: (@escaping (NCUpdateResult) -> Void)) {
        completionHandler(NCUpdateResult.newData)
    }
}

File: tests/common/TestProjects/MyTodayExtension/TodayViewController.cs
using System;
using System.Drawing;

using NotificationCenter;
using Foundation;
using Social;
using UIKit;

namespace MyTodayExtension {
	public partial class TodayViewController : UIViewController, INCWidgetProviding {
		public TodayViewController (IntPtr handle) : base (handle)
		{
		}

		public override void DidReceiveMemoryWarning ()
		{
			// Releases the view if it doesn't have a superview.
			base.DidReceiveMemoryWarning ();

			// Release any cached data, images, etc that aren't in use.
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();

			// Do any additional setup after loading the view.
		}

		[Export ("widgetPerformUpdateWithCompletionHandler:")]
		public void WidgetPerformUpdate (Action<NCUpdateResult> completionHandler)
		{
			// Perform any setup necessary in order to update the view.

			// If an error is encoutered, use NCUpdateResultFailed
			// If there's no update required, use NCUpdateResultNoData
			// If there's an update, use NCUpdateResultNewData

			completionHandler (NCUpdateResult.NewData);
		}
	}
}
